Sources/Widgets/Widgets3D/EllipseWidget/behavior.js

```diff
@@ -9,38 +8,26 @@ export default function widgetBehavior(publicAPI, model) {
 
   model.classHierarchy.push('vtkEllipseWidgetProp');
 
-  publicAPI.setBounds = (bounds) => {
-    if (superClass.setBounds) {
-      superClass.setBounds(bounds);
+  publicAPI.setCorners = (point1, point2) => {
+    if (superClass.setCorners) {
+      superClass.setCorners(point1, point2);
     }
 
-    const center = vtkBoundingBox.getCenter(bounds);
-    const scale3 = vtkBoundingBox.computeScale3(bounds);
-
-    model.shapeHandle.setOrigin(center);
-    model.shapeHandle.setScale3(scale3);
-  };
-
-  publicAPI.setBoundsFromRadius = (center, pointOnCircle) => {
-    const radius = vec3.distance(center, pointOnCircle);
-
-    publicAPI.setBounds([
-      center[0] - radius,
-      center[0] + radius,
-      center[1] - radius,
-      center[1] + radius,
-      center[2] - radius,
-      center[2] + radius,
-    ]);
-  };
-
-  publicAPI.setBoundsFromDiameter = (point1, point2) => {
     const center = [
       0.5 * (point1[0] + point2[0]),
       0.5 * (point1[1] + point2[1]),
       0.5 * (point1[2] + point2[2]),
     ];
 
-    publicAPI.setBoundsFromRadius(center, point1);
+    const d = [0, 0, 0];
+    vec3.subtract(d, point2, center);
+
+    const horizontal = model.shapeHandle.getRight();
+    const vertical = model.shapeHandle.getUp();
+    const sh = vec3.dot(d, horizontal);
```

Comment:
rename `sh` into `horizontalScale` (same for `sv`)

Reply:
will rename to rightComponent and upComponent